test/lit/passes/translate-eh-old-to-new.wast

```diff
@@ -0,0 +1,1421 @@
+  ;; CHECK-NEXT:       (i32.const 0)
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $delegate-target-caller-concrete (result i32)
+    ;; A try-delegate targets the caller whose type is concrete
+    (call $foo)
+    (try (result i32)
+      (do
+        (i32.const 0)
+      )
+      (delegate 0)
+    )
+  )
+
+  ;; CHECK:      (func $delegate-nested-more (type $1)
+  ;; CHECK-NEXT:  (block $outer3
+  ;; CHECK-NEXT:   (block $catch_all4
+  ;; CHECK-NEXT:    (try_table (catch_all $catch_all4)
+  ;; CHECK-NEXT:     (throw_ref
+  ;; CHECK-NEXT:      (block $l00 (result exnref)
+  ;; CHECK-NEXT:       (block $outer1
+  ;; CHECK-NEXT:        (block $catch_all2
+  ;; CHECK-NEXT:         (try_table (catch_all $catch_all2)
+  ;; CHECK-NEXT:          (try_table (catch_all_ref $l00)
+  ;; CHECK-NEXT:           (call $foo)
+  ;; CHECK-NEXT:          )
+  ;; CHECK-NEXT:         )
+  ;; CHECK-NEXT:         (br $outer1)
+  ;; CHECK-NEXT:        )
+  ;; CHECK-NEXT:        (nop)
+  ;; CHECK-NEXT:       )
+  ;; CHECK-NEXT:       (br $outer3)
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $delegate-nested-more
+    (try $l0
+      (do
+        (try
+          (do
+            (try
+              (do
+                (call $foo)
+              )
+              (delegate $l0)
+            )
+          )
+          (catch_all)
+        )
+      )
+      (catch_all)
+    )
+  )
+
+  ;; CHECK:      (func $delegate-target-outer-try-delegate (type $1)
+  ;; CHECK-NEXT:  (throw_ref
+  ;; CHECK-NEXT:   (block $__binaryen_delegate_caller_target0 (result exnref)
+  ;; CHECK-NEXT:    (block $outer2
+  ;; CHECK-NEXT:     (throw_ref
+  ;; CHECK-NEXT:      (block $l01 (result exnref)
+  ;; CHECK-NEXT:       (try_table (catch_all_ref $l01)
+  ;; CHECK-NEXT:        (nop)
+  ;; CHECK-NEXT:       )
+  ;; CHECK-NEXT:       (br $outer2)
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (return)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $delegate-target-outer-try-delegate
+    ;; An inner try-delegate targets an outer try-delegate that targets the
+    ;; caller
+    (try $l0
+      (do
+        (try
+          (do)
+          (delegate $l0)
+        )
+      )
+      (delegate 0)
+    )
+  )
+
+  ;; CHECK:      (func $delegate-target-outer-try-delegate-concrete (type $2) (result i32)
+  ;; CHECK-NEXT:  (throw_ref
+  ;; CHECK-NEXT:   (block $__binaryen_delegate_caller_target0 (result exnref)
+  ;; CHECK-NEXT:    (return
+  ;; CHECK-NEXT:     (block $outer2 (result i32)
+  ;; CHECK-NEXT:      (throw_ref
+  ;; CHECK-NEXT:       (block $l01 (result exnref)
+  ;; CHECK-NEXT:        (br $outer2
+  ;; CHECK-NEXT:         (try_table (result i32) (catch_all_ref $l01)
```

Comment:
I'm surprised to see a single `try_table` for a testcase that has two `try`s. Is that an optimization or must it be like this for some reason?

Reply:
No it was a bug. Pushed a fix, thanks.

Reply:
https://github.com/WebAssembly/binaryen/pull/6210/commits/9eb535bafef7c5636b7ebead3b85fc128d27a5fb